app/main.py

```diff
@@ -1,2 +1,136 @@
+        else:
+            while current_node:
+                if current_node.key == key:
+                    current_node.value = value
+                    return
+                if current_node.next is None:
+                    current_node.next = Node(key, value, node_hash)
+                    self.size += 1
+                    break
+                current_node = current_node.next
+
+        if self.size / self.capacity > self.load_factor:
+            self._resize()
+
+    def __getitem__(self, key: Any) -> Any:
+        node_hash = hash(key)
+        index = self._hash(node_hash)
+        current_node = self.table[index]
+
+        while current_node:
+            if current_node.key == key:
+                return current_node.value
+            current_node = current_node.next
+
+        raise KeyError(f"Key '{key}' not found")
+
+    def __len__(self) -> int:
+        return self.size
+
+    def _hash(self, node_hash: int) -> int:
+        return node_hash % self.capacity
+
+    def _resize(self) -> None:
+        old_table = self.table
+        self.capacity *= 2
+        self.table = [None] * self.capacity
+        self.size = 0
+
+        for current_node in old_table:
+            while current_node:
+                self[current_node.key] = current_node.value
+                current_node = current_node.next
+
+    def clear(self) -> None:
+        self.table = [None] * self.capacity
+        self.size = 0
+
+    def __delitem__(self, key: Any) -> None:
+        node_hash = hash(key)
+        index = self._hash(node_hash)
+        current_node = self.table[index]
+        prev_node = None
+
+        while current_node:
+            if current_node.key == key:
+                if prev_node is None:
+                    self.table[index] = current_node.next
+                else:
+                    prev_node.next = current_node.next
+                self.size -= 1
+                return
+            prev_node = current_node
+            current_node = current_node.next
+
+        raise KeyError(f"Key '{key}' not found")
+
+    def get(self, key: Any, default: Any = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def pop(self, key: Any, default: Any = None) -> Any:
+        node_hash = hash(key)
+        index = self._hash(node_hash)
+        current_node = self.table[index]
+        prev_node = None
+
+        while current_node:
+            if current_node.key == key:
+                value = current_node.value
+                if prev_node is None:
+                    self.table[index] = current_node.next
+                else:
+                    prev_node.next = current_node.next
+                self.size -= 1
+                return value
+            prev_node = current_node
+            current_node = current_node.next
+
+        if default is not None:
+            return default
+
+        raise KeyError(f"Key '{key}' not found")
```

Comment:
The `pop` method correctly raises a KeyError with a message when the key is not found and no default is provided, which is consistent with the code style requirement.